src/api/chat.ts

```diff
@@ -10,24 +11,55 @@ export interface SendVenueMessageProps {
 export const sendVenueMessage = async ({
   venueId,
   message,
-}: SendVenueMessageProps) =>
-  await firebase
+}: SendVenueMessageProps): Promise<void | firebase.firestore.DocumentReference<firebase.firestore.DocumentData>> =>
```

Comment:
Done in https://github.com/sparkletown/sparkle/pull/1374/commits/7e51ec9c79c9ca6626435c644f2ff6cf15d8aaca

Since .add() returns the data we don't use, I had to add `noop`

Reply:
Instead of the `noop` could we just make it return the type of the data we added?

Something like `.then(docSnapshot => docSnapshot.data())` and then make the ultimate return type `Promise<VenueChatTypeThingHere>` or similar.

While we don't make use of the returned data currently, it doesn't make sense to arbitrarily `noop` it from the `api/*` layer.

Reply:
Yes, but in this case I will not be able to make a single `SendMessage` type because they will have different return types. And since we don't use that data, what's a better approach?

Reply:
Hrmm.. Don't all of the chat types have a common 'ancestor' that they extend from? If so, I would probably use that. If not, I would probably make a 'any' type like we do for other things or similar.